FetchMigration/python/index_operations.py

```diff
@@ -106,3 +112,54 @@ def doc_count(indices: set, endpoint: EndpointInfo) -> IndexDocCount:
         return IndexDocCount(total, count_map)
     except RuntimeError as e:
         raise RuntimeError(f"Failed to fetch doc_count: {e!s}")
+
+
+def __fetch_templates(endpoint: EndpointInfo, path: str, root_key: str, factory) -> set:
+    url: str = endpoint.add_path(path)
+    # raises RuntimeError in case of any request errors
```

Comment:
Thanks for this comment. I just learned that Python3 supports [exception chaining](https://docs.python.org/3/tutorial/errors.html#exception-chaining) so i'll incorporate that here instead of stringifying the underlying exception.

> What happens if there's an exception? If looks like it propagates up?

That's correct. This was an intentional decision to fail the metadata migration if templates could not be fetched (and therefore migrated)

Reply:
Exception chaining added with [this commit](https://github.com/opensearch-project/opensearch-migrations/pull/477/commits/aaf37611e31f1225a29113d093f72df457eb0c1f)